src/main/java/net/kaczmarzyk/spring/data/jpa/domain/JoinFetch.java

```diff
@@ -86,7 +87,11 @@ public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query, CriteriaBuild
 					}
 
 					Fetch<?,?> joinFetch = evaluatedJoinFetchForGivenAlias.fetch(path, joinType);
-					context.putEvaluatedJoinFetch(alias, joinFetch);
+					context.putEvaluatedJoinFetch(alias, evaluatedJoinFetchForGivenAlias);
```

Comment:
In my opinion `context.putEvaluatedJoinFetch(alias, evaluatedJoinFetchForGivenAlias);` is redundant

Reply:
Yes, you are right. Missed that one. Removed the line